src/metrics-gatherer.ts

```diff
@@ -162,46 +183,51 @@ export class MetricsGatherer {
 	// check that a metric is of the given type(s)
 	public checkMetricType(name: string, kinds: string[]) {
 		try {
-			return kinds.includes(this.kinds[name]);
+			return kinds.includes(this.meta[name].kind);
 		} catch (e) {
 			this.err(e);
 		}
 	}
 
+	public getMetric(name: string): prometheus.Metric | undefined {
+		if (this.meta[name]) {
+			return this.metrics[this.meta[name].kind][name];
+		}
+	}
+
+	public exists(name: string) {
+		return this.getMetric(name) != null;
+	}
+
 	// used declaratively to ensure a given metric of a certain kind exists,
 	// given some custom params to instantiate it if absent
-	public ensureExists(
+	private ensureExists(
+		kind: Kind,
 		name: string,
-		kind: string,
-		labels: LabelSet = {},
```

Comment:
I'm guessing this was reverted accidentally as part of a rebase, but I very much want to keep this behaviour as it avoids a *lot* of unnecessary `Object.keys` calls which whilst being cheap individually adds up due to the sheer number of calls, which will only increase as we add more metrics